blocks/library/shortcode/index.js

```diff
@@ -39,7 +39,10 @@ registerBlockType( 'core/shortcode', {
 				// letters, but numbers and underscores should work fine too.
 				// Be wary of using hyphens (dashes), you'll be better off not
 				// using them." in https://codex.wordpress.org/Shortcode_API
-				tag: '[a-z0-9_-]+',
+				// Require that the first character be a letter. This notably
+				// prevents footnote markings ([1]) from being caught as
+				// shortcodes.
+				tag: '[a-z][a-z0-9_-]*',
```

Comment:
This whole line looks a bit add to me... maybe I missed some conversation. There is now a fallback block block for unhandled shortcodes? What if they are inline shortcodes? I'll maybe open a new ticket, sorry. :)

Reply:
Correct, see #3610. 

Inline shortcodes are definitely a problem. I’d be grateful if you could weigh in on https://github.com/WordPress/gutenberg/issues/3806#issuecomment-349502360 !